presto-orc/src/main/java/com/facebook/presto/orc/writer/ColumnWriterFactory.java

```diff
@@ -0,0 +1,88 @@
+package com.facebook.presto.orc.writer;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.orc.ColumnWriterOptions;
+import com.facebook.presto.orc.DwrfEncryptionInfo;
+import com.facebook.presto.orc.OrcEncoding;
+import com.facebook.presto.orc.metadata.MetadataWriter;
+import com.facebook.presto.orc.metadata.OrcType;
+import org.joda.time.DateTimeZone;
+
+import java.util.List;
+
+import static com.facebook.presto.orc.writer.ColumnWriters.createColumnWriter;
+
+public class ColumnWriterFactory
```

Comment:
Why do we need this class to be implemented using a builder pattern instead of passing everything through the constructor? Is there a good reason for that?